src/transformState.js

```diff
@@ -5,7 +5,28 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
-}
+  let tempState = state;
+
+  for (const singleAction of actions) {
+    switch (singleAction.type) {
+      case 'addProperties':
+        tempState = Object.assign(tempState, singleAction.extraData);
+        break;
+
+      case 'removeProperties':
+        singleAction.keysToRemove.forEach(el => {
+          delete tempState[el];
+        });
+        break;
+
+      case 'clear':
+        for (const el in tempState) {
+          delete tempState[el];
+        };
```

Comment:
Add default case for error handling
_[CODE STYLE]: switch/case should always have default case for error handling._